src/prefect/cli/flow_run.py

```diff
@@ -209,13 +231,12 @@ async def logs(
                 offset=offset,
                 sort=(
                     schemas.sorting.LogSort.TIMESTAMP_DESC
-                    if reverse
+                    if reverse or tail
                     else schemas.sorting.LogSort.TIMESTAMP_ASC
                 ),
```

Comment:
We want offset to be 0 as we iterate through the first time with `tail` because we are retrieving the logs in TIMESTAMP DESC order. If user specified `flow-run logs --tail --num_logs 250` for example, your logic below would change offset to 50:
offset = max(0, user_specified_num_logs - LOGS_DEFAULT_PAGE_SIZE).

In other words, the `if reverse or tail` condition you added should be sufficient and you should be able to remove the code that sets the tail to a specific limit or offset as I believe the existing code was made to be extensible to new use cases like this one.

Reply:
I believe there is a bug in the previous logic. 
`            # Update the number of logs retrieved
            num_logs_returned += num_logs_to_return_from_page

            if len(page_logs) == LOGS_DEFAULT_PAGE_SIZE:
                offset += LOGS_DEFAULT_PAGE_SIZE
            else:
                # No more logs to show, exit
                more_logs = False
`

When using the command `flow-run logs --head --num_logs 250 `to retrieve logs and assuming there are 500 logs available, the code will retrieve all 500 logs because it does not utilize the `num_logs_returned` variable. The code only stops when there are no more logs left, as indicated by the `more_logs` flag. Similarly, if the previous logic were to be used for the `--tail` command, it would first return the last 200 logs, followed by the rest of the logs in the same manner, with logs 301 to 500, 101 to 300, and finally 1 to 100.

Reply:
Not up to date on this but there's some changes to this logic happening in #9130 — would that resolve problems here?

Reply:
Hi @madkinsz Based on my understanding of #9130 it seems that the code has been modified to eliminate hard-coded variables. Additionally, a new API has been incorporated that retrieves all the logs to determine their length. However, it appears that the issue of limiting the logs has not been resolved.

Reply:
I've put up a PR for that bug: https://github.com/PrefectHQ/prefect/pull/9314

Reply:
Once #9314 is merged I will make changes accordingly

Reply:
Hi @serinamarie  I've updated the PR, but we still require the offset for the tail flag. To clarify, if there are 500 logs and I want to retrieve the last 250 logs using the command flow-run logs --tail --num_logs 250, the offset should not start from 0. If it does, the command will return the last 200 logs (logs 301-500) followed by logs 251-300. Although the logs themselves are correct, the sequence will be inaccurate.

Reply:
Ah, you're right. The code looks clearer now, thanks 👍